src/core/link-to-dfn.js

```diff
@@ -128,43 +136,35 @@ function collectDfns(title) {
 }
 
 /**
- * @param {import("./utils.js").LinkTarget} target
  * @param {HTMLAnchorElement} anchor
- * @param {CaseInsensitiveMap} titleToDfns
+ * @param {import("./utils.js").LinkTarget} target
+ * @param {ReturnType<typeof mapTitleToDfns>} titleToDfns
  */
-function findLinkTarget(target, anchor, titleToDfns) {
+function processAnchor(anchor, target, titleToDfns) {
+  let localMatchFound = false;
   const { linkFor } = anchor.dataset;
-  if (
-    !titleToDfns.has(target.title) ||
-    !titleToDfns.get(target.title).get(target.for)
-  ) {
-    return false;
-  }
   const dfn = titleToDfns.get(target.title).get(target.for);
   if (dfn.dataset.cite) {
     anchor.dataset.cite = dfn.dataset.cite;
+    localMatchFound = true;
   } else if (linkFor && !titleToDfns.get(linkFor)) {
-    possibleExternalLinks.push(anchor);
+    localMatchFound = false;
```

Comment:
This is weird. How about `noLocalMatch` and set it true if not found, and then return `!noLocalMatch`?

Reply:
> This is weird.

Yes! It's meant to be a noop here, but if we skip this condition, tests fail :tada: Decided setting it to false was better than a comment? Maybe comment was better option.

`noLocalMatch` is a lot of negations IMO.

Reply:
>noLocalMatch is a lot of negations IMO.

I think it's no problem if it makes cleaner code and not causes confusion.